### PR TITLE
add autoresume

### LINKED ISSUE #623: Autoresume from "logdir/last_full.pth"
In some systems, like slurm, task is time limited. Usually it is about 10-20 hours. Large tasks, like ImageNet, require running several small tasks. This feature helps automate this process.

### PR DESCRIPTION
**Pull request by AndreySheka**
## Description
close #623 

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have read I need to click 'Login as guest' to see Teamcity build logs
- [x] I have checked the code-style using `make check-codestyle`.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Review by bagxi — CHANGES_REQUESTED** [review]
Please use double-quotes

File: catalyst/dl/scripts/run.py [hunk 1]
```diff
@@ -44,18 +44,25 @@ def build_args(parser: ArgumentParser):
         metavar="PATH",
         help="path to latest checkpoint"
     )
+    parser.add_argument(
```
Comment (bagxi) [inline catalyst/dl/scripts/run.py]: Could you replace `parser.add_argument(action="store_true", ...)` with `utils.boolean_flag(...)`?

**Review by bagxi — CHANGES_REQUESTED** [review]

File: catalyst/dl/scripts/run.py (hunk 1 above)
Comment (Scitator) [reply catalyst/dl/scripts/run.py]: ➕

File: catalyst/utils/config.py [hunk 2]
```diff
@@ -254,6 +254,15 @@ def parse_config_args(*, config, args, unknown_args):
...
 
+    if "autoresume" in config["args"] and \
+            "logdir" in config["args"] and \
+            "resume" not in config["args"]:
+        log_dir = config["args"]["logdir"]
+        checkpoint_filename = os.path.join(
+            log_dir, "checkpoints", "last_full.pth"
+        )
+        if os.path.isfile(checkpoint_filename):
+            config["resume"] = checkpoint_filename
```
Comment (Scitator) [inline catalyst/utils/config.py]: does this feature supports loading checkpoint for Stage2/etc? the case is the following
1. you are training multi-stage pipeline on slurm
2. stage1 goes okay
3. stage2 breaks
4. you are loading the checkpoint from stage2 to resume the training

which stage would it become training?

Comment (AndreySheka) [reply catalyst/utils/config.py]: Slurm is not important in this case. I stopped training on stage1 and stage2, and it resumed fine with pull request 629. This pull request is more like syntactic sugar.

File: catalyst/dl/scripts/run.py (hunk 1 above)
Comment (AndreySheka) [reply catalyst/dl/scripts/run.py]: done

File: catalyst/utils/config.py [hunk 3]
```diff
@@ -254,6 +254,15 @@ def parse_config_args(*, config, args, unknown_args):
                 continue
             config["args"][key] = value
 
+    if "autoresume" in config["args"] and \
+            "logdir" in config["args"] and \
+            "resume" not in config["args"]:
```
Comment (Scitator) [inline catalyst/utils/config.py]: don't you mean
```
config.get("autoresume", False) and config.get("logdir", None) is not None and config.get("resume", None) is None
```
?

**Review by Scitator — CHANGES_REQUESTED** [review]

File: catalyst/utils/config.py (hunk 3 above)
Comment (AndreySheka) [reply catalyst/utils/config.py]: fixed

File: catalyst/utils/config.py [hunk 4]
```diff
@@ -254,6 +254,13 @@ def parse_config_args(*, config, args, unknown_args):
                 continue
             config["args"][key] = value
 
+    if safitty.get(config, "args", "autoresume", default=False) and \
+            safitty.get(config, "args", "logdir") is not None and \
+            safitty.get(config, "args", "resume") is None:
+        log_dir = Path(safitty.get(config, "args", "logdir"))
+        checkpoint_filename = log_dir / "checkpoints" / "last_full.pth"
+        if checkpoint_filename.is_file():
+            config["resume"] = str(checkpoint_filename)
```
Comment (Scitator) [inline catalyst/utils/config.py]: don't you mean `config["args"]["resume"] = ...`?

File: catalyst/utils/config.py [hunk 5]
```diff
@@ -254,6 +254,13 @@ def parse_config_args(*, config, args, unknown_args):
                 continue
             config["args"][key] = value
 
+    if safitty.get(config, "args", "autoresume", default=False) and \
+            safitty.get(config, "args", "logdir") is not None and \
+            safitty.get(config, "args", "resume") is None:
+        log_dir = Path(safitty.get(config, "args", "logdir"))
```
Comment (Scitator) [inline catalyst/utils/config.py]: because of the codestyle, I think `logdir` would be better naming

File: catalyst/utils/config.py [hunk 6]
```diff
@@ -254,6 +254,13 @@ def parse_config_args(*, config, args, unknown_args):
...
+        checkpoint_filename = log_dir / "checkpoints" / "last_full.pth"
+        if checkpoint_filename.is_file():
```
Comment (Scitator) [inline catalyst/utils/config.py:262]: btw, does this function also checks, that file exists?

**Review by Scitator — CHANGES_REQUESTED** [review]

File: catalyst/utils/config.py (hunk 6 above)
Comment (AndreySheka) [reply catalyst/utils/config.py:262]: From docs:
Path.is_file()
Return True if the path points to a regular file (or a symbolic link pointing to a regular file), False if it points to another kind of file.

False is also returned if the path doesn’t exist or is a broken symlink; other errors (such as permission errors) are propagated.

File: catalyst/utils/config.py (hunk 5 above)
Comment (AndreySheka) [reply catalyst/utils/config.py]: fixed

File: catalyst/utils/config.py (hunk 4 above)
Comment (AndreySheka) [reply catalyst/utils/config.py]: fixed

**Pull request merged**